src/message.rs

```diff
@@ -145,8 +149,12 @@ impl MidiMsg {
                 }
                 0xF => {
                     if b & 0b00001111 == 0 {
+                        #[cfg(not(feature = "no_sysex"))]
```

Comment:
I would separate out the logic here using the `cfg` macro and an `if` statement.

Reply:
Not sure I'm following, sorry... However, I did realize you can disable entire blocks with the `cfg` macro like this, if you think that's acceptable

```
#[cfg(not(feature = "no_sysex"))]
 {
    let (msg, len) = SystemExclusiveMsg::from_midi(m, ctx)?;
    Ok((Self::SystemExclusive { msg }, len))
}
#[cfg(feature = "no_sysex")]
Err(ParseError::Invalid(format!("Got sysex message but the midi-msg crate was built with no_sysex.")))
```

Reply:
I was meaning something like this:

```
if cfg!(feature = "no_sysex") {
    Err(ParseError::Invalid(format!("Got sysex message but the midi-msg crate was built with no_sysex.")))
} else {
    let (msg, len) = SystemExclusiveMsg::from_midi(m, ctx)?;
    Ok((Self::SystemExclusive { msg }, len)
}
```

IMO, this makes the intent a little more clear: i.e. we have a true branch-point in behaviour.

Reply:
Correct me if I'm wrong, but it seems like the `if cfg!`solution leads to both `if` branches being compiled and does not offer the conditional compilation we're after.

Reply:
Using `cfg!` in a branching statement like that causes two things to happen when the code is compiled:

First macros are expanded, and `cfg!(...)` gets replaced with a boolean constant. Later the compiler backend removes branches that cannot be hit, so you wind up not paying a runtime cost (and thus it acts exactly like conditional compilation).

Reply:
Maybe I'm missing something obvious, but when I use `if cfg!(...` and run `cargo build` with the `no_sysex` feature enabled, the build fails with `failed to resolve: use of undeclared type SystemExclusiveMsg`.

Reply:
Nope, that's my bad. The branch removal doesn't happen until well after symbols are resolved, so it indeed can't be used like that if you're using a type that doesn't exist in one branch 🤦